vermouth/molecule.py

```diff
@@ -681,16 +681,51 @@ def merge_molecule(self, molecule):
                 'This molecule has nrexcl={}, while the other has nrexcl={}.'
                 .format(self.nrexcl, molecule.nrexcl)
             )
-        if self.nodes():
+        # simple just computes and offset and renumbers all resid
+        # this works well for many molecules but especially not
+        # for proteins where for example gaps in resid are common
+        if self.nodes() and offset_handling == 'simple':
             # We assume that the last id is always the largest.
             last_node_idx = max(self)
             offset = last_node_idx
             residue_offset = self.nodes[last_node_idx].get('resid', 1)
             offset_charge_group = self.nodes[last_node_idx].get('charge_group', 1)
+        # protein is more advanced and checks the intersection of chain and resid
+        # between the two merging molecules. If chains are all different, resids
```

Comment:
This is also a valid approach for DNA and arbitrary linear polymers, no? Anything for which a sane chain can be identified